src/mask.js

```diff
@@ -144,11 +146,10 @@ angular.module('ui.mask', [])
                                 // to be out-of-sync with what the controller's $viewValue is set to.
                                 controller.$viewValue = value.length ? maskValue(value) : '';
                                 controller.$setValidity('mask', isValid);
-                                if (isValid) {
-                                    return modelViewValue ? controller.$viewValue : value;
-                                } else {
-                                    return undefined;
-                                }
+
+                                if (modelViewValue) value = controller.$viewValue;
+                                if (!value.length) return undefined;
+                                if (isValid || linkOptions.unsafe) return value;
```

Comment:
Sure but it doesn't hurt to be explicit so it doesn't get lost somewhere in the future if other functionality is added.